streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/CheckpointProcessor.java

```diff
@@ -41,9 +44,16 @@
 @Component
 public class CheckpointProcessor {
 
+  private static final Byte DEFAULT_FLAG_BYTE = Byte.valueOf("0");
+  private static final Integer SAVEPOINT_CACHE_HOUR = 1;
+
   private final Cache<String, Long> checkPointCache =
       Caffeine.newBuilder().expireAfterAccess(1, TimeUnit.DAYS).build();
 
+  /** Cache to store the savepoint path if be stored in the db. */
+  private final Cache<String, Byte> savepointedCache =
```

Comment:
Why need the `savepointedCache`? Why `checkPointCache` isn't enough?

Reply:
1.  I moved the store logic for savepoint here in order to uniform data caliber, which could supply the savepoint chk-id field.
The chk-id is meaningful to adapt the store logic of checkpoints.
2. For your question: We prefer to store all the completed savepoints and checkpoints. However, if the savepoint store-logic follow `checkpoint`, here would be some bug-case. If the triggered savepoint is completed after a completed checkpoint whose chk-id is bigger than the savepoint, the savepoint will be discard  based on the current logic.     So, the `savepointedCache` was introduced, If the chk-id  of the completed savepoint is smaller than the latest stored checkpoint-id, we just store the savepoint whose triggertime within `savepointCache` .

Reply:
Thanks for your quick clarification.

> However, if the savepoint store-logic follow checkpoint, here would be some bug-case. If the triggered savepoint is completed after a completed checkpoint whose chk-id is bigger than the savepoint, the savepoint will be discard based on the current logic. 

As I understand, you have ordered the checkpoint by chk-id, so the bug cannot be happened when `execution.checkpointing.max-concurrent-checkpoints == 1` due to the `checkpoint/savepoint` with a smaller `chk-id` must be completed earlier than the checkpoint with a larger `chk-id`, right? 

Are you worried about `execution.checkpointing.max-concurrent-checkpoints > 1`?

And I mean the code is a bit complicated right now, not sure if it's necessary.

Reply:
> Are you worried about `execution.checkpointing.max-concurrent-checkpoints > 1`?

Yes, the introduced lines is used to  adapt for the setting case only.